### PR TITLE
Disable the JpaUserFederatedStorageProvider when map storage is enabled

### LINKED ISSUE #12895: Investigate why `userFederatedStorage` is not failing
See https://github.com/keycloak/keycloak/pull/12865#discussion_r912887728

### PR DESCRIPTION
**Pull request by ahus1**
Closes #12895

### CODE REVIEW

**Comment by ahus1** [comment]
@sguilhen - the reason was that I was still fixing some broken tests this morning, and then there was an unstable test that now switched to green apparently. 

And then I pushed the wrong commit, and now the build is starting again.

I'm a bit hesitant about the changes in `ExportUtils` and `MultipleStepsExportProvider` - what do you think about this not exporting federated users when there is no federation?

**Comment by sguilhen** [comment]
> I'm a bit hesitant about the changes in `ExportUtils` and `MultipleStepsExportProvider` - what do you think about this not exporting federated users when there is no federation?

I would say the changes make sense when the provider is disabled, but perhaps we should wait for @mhajas opinion on this.

File: model/legacy-services/src/main/java/org/keycloak/exportimport/util/MultipleStepsExportProvider.java [hunk 1]
```diff
@@ -84,7 +84,11 @@ protected void runExportImportTask(KeycloakSession session) throws IOException {
                 // Count total number of users
                 if (!exportUsersIntoRealmFile) {
                     usersHolder.totalCount = session.users().getUsersCount(realm, true);
-                    federatedUsersHolder.totalCount = UserStorageUtil.userFederatedStorage(session).getStoredUsersCount(realm);
+                    if (UserStorageUtil.userFederatedStorage(session) != null) {
```
Comment (mhajas) [inline model/legacy-services/src/main/java/org/keycloak/exportimport/util/MultipleStepsExportProvider.java:87]: Can this be `null`? I see that this is part of `legacy-services` module. Isn't it true that when this module is used legacy storage is used? Is this causing problems in the testsuite?

**Review by mhajas — COMMENTED** [review]
I would say the solution is correct. However, I added one question

**Review by mhajas — COMMENTED** [review]
LGTM @ahus1

**Review by mhajas — APPROVED** [review]

**Pull request merged**